core/src/main/java/org/axonframework/eventsourcing/eventstore/AbstractEventStore.java

```diff
@@ -130,4 +130,15 @@ public void storeSnapshot(DomainEventMessage<?> snapshot) {
     protected EventStorageEngine storageEngine() {
         return storageEngine;
     }
+
+    @Override
+    public Optional<Long> lastSequenceNumberFor(String aggregateIdentifier) {
+        Optional<Long> highestStaged = stagedDomainEventMessages(aggregateIdentifier)
```

Comment:
Suggestion: you could rewrite it to:
```
return stagedDomainEventMessages(aggregateIdentifier)
          .map(DomainEventMessage::getSequenceNumber)
          .max(Long::compareTo)
          .orElse(storageEngine.lastSequenceNumberFor(aggregateIdentifier));
```

Reply:
That doesn't work, because the method needs to return an optional. `orElse` doesn't return an `Optional`, but the instance it contains.

Reply:
Ah, of course. Disregard my suggestions.